src/nvim/macros.h

```diff
@@ -133,12 +133,12 @@
 /* Get the length of the character p points to */
 # define MB_PTR2LEN(p)          (has_mbyte ? (*mb_ptr2len)(p) : 1)
 /* Advance multi-byte pointer, skip over composing chars. */
-# define mb_ptr_adv(p)      p += has_mbyte ? (*mb_ptr2len)(p) : 1
+# define mb_ptr_adv(p)      p += has_mbyte ? (*mb_ptr2len)((char_u *)p) : 1
 /* Advance multi-byte pointer, do not skip over composing chars. */
 # define mb_cptr_adv(p)     p += \
   enc_utf8 ? utf_ptr2len(p) : has_mbyte ? (*mb_ptr2len)(p) : 1
 /* Backup multi-byte pointer. Only use with "p" > "s" ! */
-# define mb_ptr_back(s, p)  p -= has_mbyte ? ((*mb_head_off)(s, p - 1) + 1) : 1
+# define mb_ptr_back(s, p)  p -= has_mbyte ? ((*mb_head_off)((char_u *)s, (char_u *)p - 1) + 1) : 1
```

Comment:
I think it's ok, until those functions are refactored too.
What I would do is taking the chance to correct al macros in this block whose body is not surrounded by parentheses. That can lead to nasty bugs.
When I face something like this (i.e., something that is unrelated to current task, but that is easy to do, and I don't want to leave it for later (not to forget)), I do a separated commit with title "Passing by: <whatever is done>".
You can do that in this case, or take a note for a spun-off PR, if you want.